src/main/java/org/jenkinsci/plugins/ownership/integrations/securityinspector/OwnerFilter.java

```diff
@@ -84,36 +93,62 @@ public OwnerFilter(@Nonnull StaplerRequest req)
             includeRegex = null;
             includePattern = null;
         }
+        
+        if (req.getParameter("usefolder") != null) {
+            report4folder = req.getParameter("selectedFolder");
+        } else {
+            report4folder = null;
+        }
     }
     
     @Nonnull
     @Restricted(NoExternalUse.class)
     public List<TopLevelItem> doFilter(User owner) {
         
         final Jenkins jenkins = JenkinsHelper.getInstanceOrFail();
-        final List<Item> allItems = jenkins.getAllItems(Item.class);
+        final List<Item> allItems;
+        
+        if (report4folder != null) {
+            Item folder = jenkins.getItem(report4folder);
+            if (folder instanceof ItemGroup) {
+                Collection<Item> items = ((ItemGroup)folder).getItems();
+                allItems = new ArrayList<>(items.size());
+                for (Item item : items) {
+                    allItems.add(item);
+                }
+                allItems.add(folder);
+            } else {
+                throw new IllegalStateException(report4folder + " is not an ItemGroup");
```

Comment:
If such issue happens, log it and return an empty list. It is better to void runtime exception in the code where possible